### PR TITLE
T-172 Report page and Reports sending tasks

### PR DESCRIPTION
**Pull request by UvgenGen**
[T-184](https://youtrack.raccoongang.com/issue/T-184) Extended report divided by roles
[T-172](https://youtrack.raccoongang.com/issue/T-172) Report page
[T-176](https://youtrack.raccoongang.com/issue/T-176) Report composition
[T-180](https://youtrack.raccoongang.com/issue/T-180) Parents and Teachers Reports

### CODE REVIEW

File: common/djangoapps/tedix_ro/views.py [hunk 1]
```diff
@@ -16,37 +15,91 @@
...
+
+    user = User.objects.prefetch_related("groups").get(id=user.id)
+    if request.user.id != user.id:
+        # refetch the course as the assumed student
+        course = get_course_with_access(user, 'load', course_key, check_if_enrolled=True)
+
+    report_data = []
+    modulestore_course = modulestore().get_course(course_key)
+    if user.is_superuser:
+        for student in User.objects.filter(courseenrollment__course_id=course.id).select_related('profile'):
```
Comment (sendr) [inline common/djangoapps/tedix_ro/views.py]: student or teacher or just user? check please naming

Comment (sendr) [reply common/djangoapps/tedix_ro/views.py]: maybe worth adding additional check to avoid staffs here?

Comment (sendr) [reply common/djangoapps/tedix_ro/views.py]: something like `is_staff=False`
or I do know something?

File: common/djangoapps/tedix_ro/views.py [hunk 2]
```diff
@@ -16,37 +15,91 @@
...
+            header, user_data = report_data_preparation(student, modulestore_course, course_key)
+            report_data.append(user_data)
+    elif user.is_staff and hasattr(user, 'instructorprofile'):
+        for student_profile in user.instructorprofile.students.filter(user__courseenrollment__course_id=course.id).select_related('user__profile'):
```
Comment (sendr) [inline common/djangoapps/tedix_ro/views.py]: what length of this line?

File: common/djangoapps/tedix_ro/views.py [hunk 3]
```diff
@@ -186,8 +239,8 @@ def get_serializer_class(self):
 
 
 class VideoLessonViewSet(viewsets.ModelViewSet):
-    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
-    permission_classes = (permissions.IsAuthenticated,)
+    # authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
```
Comment (sendr) [inline common/djangoapps/tedix_ro/views.py]: do we really need to comment this?

File: common/djangoapps/tedix_ro/templates/includes/student_extended_report.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+%for user_data in report_data:
+    <h2 class="hd hd-2 progress-certificates-title">
+        <h4 class="hd hd-4">
+            <p>${_("Student name")}: <span style="color: red;">${ user_data['full_name'] }</span><p>
+            <p>${_("Activity report for lesson")}: ${ course_name }<p>
+            <p>${_("Lesson completed")}: <span style="color: ${'green' if user_data['completion'] == 1 else 'red'};">${'Yes' if user_data['completion'] == 1 else 'No'}</span><p>
+            <p>${_("Number of question completed from first attempt")}:<span style="color: green"> ${ user_data['percent'] }% (${user_data['count_answer_first_attempt']} out of ${len(user_data['questions'])})</span><p>
+        </h4>
+    </h2>
+    <section class="chapters">
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply common/djangoapps/tedix_ro/templates/includes/student_extended_report.html]: closing tag on line 28

File: common/djangoapps/tedix_ro/utils.py [hunk 5]
```diff
@@ -1,10 +1,112 @@
...
+    student_course_due_date = StudentCourseDueDate.objects.filter(student__user=user, course_id=course.id).first()
+    due_date = student_course_due_date.due_date if student_course_due_date else None
+    report_data = {
+        'student_name': user.profile.name or user.username,
+        'lesson': course.display_name,
+        'completion': completion,
+        'count_answer_first_attempt': count_answer_first_attempt,
+        'count_questions': count_questions,
+        'percent': (count_answer_first_attempt * 100 / count_questions) if count_questions else 100,
+        'due_date': format_datetime(due_date, "yy.MM.dd hh:mm a", locale='en'),
```
Comment (sendr) [inline common/djangoapps/tedix_ro/utils.py]: what if there is no due_date?

**Pull request merged**